src/main/java/chess/domain/piece/Pawn.java

```diff
@@ -0,0 +1,67 @@
+package chess.domain.piece;
+
+import chess.domain.point.Point;
+import chess.domain.point.Rank;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public final class Pawn extends Piece {
+
+    private static final int TWO_RANK = 2;
+    private static final Map<Team, Pawn> POOL = Map.of(
+            Team.WHITE, new Pawn(Team.WHITE),
+            Team.BLACK, new Pawn(Team.BLACK)
+    );
+
+    private Pawn(Team team) {
+        super(team);
+    }
+
+    static Pawn from(Team team) {
+        return POOL.get(team);
+    }
+
+    @Override
+    public boolean isMovable(Point currentPoint, Point nextPoint, Piece target) {
+        Team team = getTeam();
+        List<Point> movablePoints = findMovablePoints(currentPoint, team);
+        return movablePoints.contains(nextPoint) &&
+                (movableDiagonal(currentPoint, nextPoint, target) || movableStraight(currentPoint, nextPoint, target));
+
+    }
+
+    private List<Point> findMovablePoints(Point currentPoint, Team team) {
+        List<Point> points = new ArrayList<>();
+        int forwardDirection = team.forwardDirection();
+
+        if (isStartPosition(currentPoint)) {
+            findMovablePoint(points, currentPoint,
+                    Direction.of(0, TWO_RANK * forwardDirection));
```

Comment:
해당 코드에선 어떤 Direction으로 가는지 직관적으로 보이는 게 훨씬 좋은데, 굳이 file과 rank를 넘겨 계산하는 방식은 이해가 어렵습니다. 
우테코는 삼항연산자 혐오집단이긴 하지만 어떤 식으로든 방향을 명시적으로 표현해주는게 나아보입니다.
```suggestion
                    forwardDirection ? TWO_UPPER : TWO_LOWER;
```

Reply:
해당 코드가 부수 작용도 많고 가독성도 떨어져 리팩토링을 많이 고민했었습니다. 남들과 다른 길을 걸으니 최적화하는데 한계가 느껴지기도 하네요. 최대한 가독성 좋게 리팩토링 해보겠습니다.